src/hooks/useBetterTrade.ts

```diff
@@ -0,0 +1,212 @@
+import { Currency, CurrencyAmount, Percent, TradeType } from '@uniswap/sdk-core'
+import { Router, Trade as V2Trade } from '@uniswap/v2-sdk'
+import { FeeAmount, Pool, Trade as V3Trade } from '@uniswap/v3-sdk'
+import { WETH9_EXTENDED } from 'constants/tokens'
+import { BigNumber } from 'ethers'
+import { useActiveWeb3React } from 'hooks/web3'
+import JSBI from 'jsbi'
+import { useEffect, useMemo, useState } from 'react'
+import { isTradeBetter } from 'utils/isTradeBetter'
+import isZero from 'utils/isZero'
+import { useGasEstimateForApproval } from './useApproveCallback'
+import { useV2RouterContract } from './useContract'
+import { usePools } from './usePools'
+import useTransactionDeadline from './useTransactionDeadline'
+
+/**
+ * Return V2 Router swap gas estimate.
+ * Not required for V3 as the routing API returns the estimates.
+ */
+function useV2SwapGasEstimate(trade: V2Trade<Currency, Currency, TradeType> | undefined) {
+  const { library } = useActiveWeb3React()
+  const deadline = useTransactionDeadline()
+  const routerContract = useV2RouterContract()
+
+  const [v2SwapGasEstimateWei, setV2SwapGasEstimate] = useState<BigNumber | undefined>()
+
+  useEffect(() => {
+    if (!library || !routerContract || !trade) return
+
+    const sampleSlippage = new Percent(5, 100)
+    const sampleRecipient = '0xab5801a7d398351b8be11c439e05c5b3259aec9b'
```

Comment:
out of curiosity what is this? also should probably be checksummed

Reply:
was Vitalik's address 😬 will most likely be removed!